client/controller/index.web.js

```diff
@@ -16,26 +16,29 @@ import Layout from 'layout';
 import LayoutLoggedOut from 'layout/logged-out';
 import { login } from 'lib/paths';
 import { makeLayoutMiddleware } from './shared.js';
-import { getCurrentUser } from 'state/current-user/selectors';
+import { isUserLoggedIn } from 'state/current-user/selectors';
 import { getImmediateLoginEmail, getImmediateLoginLocale } from 'state/immediate-login/selectors';
-import userFactory from 'lib/user';
+import { getCurrentRoute } from 'state/selectors/get-current-route';
 
 /**
  * Re-export
  */
 export { setSection, setUpLocale } from './shared.js';
 
-const user = userFactory();
+export const ReduxWrappedLayout = ( { store, primary, secondary, redirectUri } ) => {
+	const state = store.getState();
+	const currentRoute = getCurrentRoute( state );
+	const userLoggedIn = isUserLoggedIn( state );
+	let layout = <Layout primary={ primary } secondary={ secondary } />;
 
-export const ReduxWrappedLayout = ( { store, primary, secondary, redirectUri } ) => (
-	<ReduxProvider store={ store }>
-		{ getCurrentUser( store.getState() ) ? (
-			<Layout primary={ primary } secondary={ secondary } user={ user } />
-		) : (
+	if ( ! userLoggedIn || /^\/start\/user-continue\//.test( currentRoute ) ) {
```

Comment:
Moved this check from `Layout#renderMasterbar` (which was removed in this PR) https://github.com/Automattic/wp-calypso/pull/27549/files#diff-e8ab61833a6d85e586cf195046ed6f15L75

Reply:
FYI @mattwiebe since you introduced this in #27254

Reply:
@brbrr Nitpick, let's use lodash's `startsWith()` instead of the RegEx here.